src/app/inventory/store/d2-item-factory.service.ts

```diff
@@ -887,6 +918,29 @@ export function D2ItemFactory(
     };
   }
 
+  function buildMasterworkInfo(
+    sockets: DimSockets,
+    statDefs: LazyDefinition<DestinyStatDefinition>
+  ): DimMasterwork | null {
+    const progress = _.find(_.pluck(sockets.sockets, 'masterworkProgress'), (mp) => mp >= 0);
```

Comment:
Unless you expect the masterwork properties to be spread out over a bunch of different sockets, it's better to just find the one you want first, then read out the properties from it.